code/modules/food/beverages/bottler.dm

```diff
@@ -0,0 +1,413 @@
+		drink_container = new drink_container()
+		drink_container.canopened = 1
+		drink_container.flags |= OPENCONTAINER
+		drink_container.forceMove(loc)
+		containers[con_type]--
+
+/obj/machinery/bottler/proc/process_ingredients(container = 0)
+	//stop if we have ZERO ingredients (what would you process?)
+	if(!slots[1] && !slots[2] && !slots[3])
+		visible_message("<span class='warning'>There are no ingredients to process! Please insert some first.</span>")
+		return
+	//prep a container
+	var/obj/item/weapon/reagent_containers/food/drinks/cans/bottler/drink_container
+	var/con_type
+	switch(container)
+		if(1)	//glass bottle
+			con_type = "glass bottle"
+			drink_container = /obj/item/weapon/reagent_containers/food/drinks/cans/bottler/glass_bottle
+		if(2)	//plastic bottle
+			con_type = "plastic bottle"
+			drink_container = /obj/item/weapon/reagent_containers/food/drinks/cans/bottler/plastic_bottle
+		if(3)	//metal can
+			con_type = "metal can"
+			drink_container = /obj/item/weapon/reagent_containers/food/drinks/cans/bottler/metal_can
+
+	if(!con_type)
+		visible_message("<span class='warning'>Error 404: Drink Container Not Found.</span>")
+		return
+	if(!containers[con_type])
+		visible_message("<span class='warning'>Error 503: Out of [con_type]s.</span>")
+		return
+	else
+		new drink_container()
+		containers[con_type]--
+	//select and process a recipe based on inserted ingredients
+	visible_message("<span class='notice'>[src] hums as it processes the ingredients...</span>")
+	bottling = 1
+	var/datum/bottler_recipe/recipe_to_use = select_recipe()
+	if(!recipe_to_use)
+		//bad recipe, ruins the drink
+		var/contents = pick("thick goop", "pungent sludge", "unspeakable slurry", "gross-looking concoction", "eldritch abomination of liquids")
+		visible_message("<span class='warning'>The [con_type] fills with \an [contents]...</span>")
+		drink_container.reagents.add_reagent(pick("????", "toxic_slurry", "meatslurry", "glowing_slury", "fishwater"), pick(30, 50))
+		drink_container.name = "Liquid Mistakes"
+		drink_container.desc = "WARNING: CONTENTS MAY BE AWFUL, DRINK AT OWN RISK."
+	else
+		//good recipe, make it
+		visible_message("<span class='notice'>The [con_type] fills with a delicious-looking beverage!</span>")
+		drink_container.reagents.add_reagent(recipe_to_use.result, 50)
+		drink_container.name = "[recipe_to_use.name]"
+		drink_container.desc = "[recipe_to_use.description]"
+	flick("bottler_on", src)
+	spawn(45)
+		for(var/i = 1, i <= slots.len, i++)
+			qdel(slots[i])
+			slots[i] = null
+		bottling = 0
+		drink_container.forceMove(loc)
+		updateUsrDialog()
+
+/obj/machinery/bottler/attack_ai(mob/user)
+	attack_hand(user)
+
+/obj/machinery/bottler/attack_ghost(mob/user)
+	attack_hand(user)
+
+/obj/machinery/bottler/attack_hand(mob/user)
+	if(stat & BROKEN)
+		return
+	interact(user)
+
+/obj/machinery/bottler/interact(mob/user)
+	user.set_machine(src)
+	//html ahoy
+	var/dat = ""
```

Comment:
Make the tables fancier: Here, add the following:

```<style>
table, th, td {
    border: 1px solid black;
    border-collapse: collapse;
}
</style>```

Then, remove the border property from all tables